frontend/app/components/address.tsx

```diff
@@ -5,26 +5,27 @@ import { cn } from '~/utils/tw-utils';
 export interface AddressProps extends ComponentProps<'address'> {
   address: string;
   city: string;
+  country: string;
+  isCanadianAddress: boolean;
   provinceState?: string;
   postalZipCode?: string;
-  country: string;
+  apartment?: string;
   altFormat?: boolean;
 }
 
-function formatAddress(address: string, city: string, country: string, provinceState?: string, postalZipCode?: string, altFormat?: boolean) {
-  // TODO 'canada' shouldn't be hardcoded as we may deal with different values of the country field such as abbreviations
-  const isNotCanadianAddress = 'canada' !== country.toLowerCase();
-
+function formatAddress(address: string, city: string, country: string, isCanadianAddress: boolean, provinceState?: string, postalZipCode?: string, apartment?: string, altFormat?: boolean) {
   // prettier-ignore
   const lines = [`${address}`,
     `${city}${provinceState ? ` ${provinceState}` : ''}${postalZipCode ? `  ${postalZipCode}` : ''}`,
-    `${isNotCanadianAddress ? country : ''}`];
+    apartment ? `Apt ${apartment}` : '',
+    `${isCanadianAddress ? '' : country}`];
 
   // prettier-ignore
   const linesAlt = [`${address}`,
   `${city}${provinceState ? ` ${provinceState}` : ''}`,
   `${postalZipCode ? `${postalZipCode}` : ''}`,
-  `${isNotCanadianAddress ? country : ''}`];
+  apartment ? `Apt ${apartment}` : '',
```

Comment:
If we follow the Canada Post format, the apartment, suite, etc. should either appear before the street number with no abbreviation (eg. 10-123 Fake St) or after the street name if there is an abbreviation (eg. 123 Fake St Apt 10).

We also shouldn't hard-code "Apt" as there are other abbreviations for units like "Suite", "Lower", "Upper", and we wouldn't want to display something like "Apt Lower 1" if the user entered "Lower 1"

Reply:
updated

Reply:
Would it make sense to do a check on `apartment` to see if it only consists of numbers/letters (no spaces)? :shrug: 

For example if the user entered "102A" as their apartment then the address will display as "102A-123 Fake St". If the user entered "Apt 102A", the address will display as "123 Fake St Apt 102A"

Reply:
I can do that pretty easily, yes

Reply:
I added a simple regex to check for a continuous chunk of alphanumeric characters as the apartment number:  `  const formattedAddress = apartment ? (/^[a-z\d]+$/i.test(apartment) ? `${apartment}-${address}` : `${address} ${apartment}`) : address;`